src/frontends/pytorch/src/op/index_add.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/scatter_elements_update.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/subtract.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_index_add(const NodeContext& context) {
+    // aten::index_add(Tensor self, int dim, Tensor index, Tensor source, *, Scalar alpha=1) -> Tensor
+    // aten::index_add.out(Tensor self, int dim, Tensor index, Tensor source, *, Scalar alpha=1, Tensor(a!) out) ->
+    // Tensor(a!)
+    num_inputs_check(context, 5, 6);
+    auto input = context.get_input(0);
+    auto dim = context.get_input(1);
+    auto index = context.mark_node(std::make_shared<v0::Convert>(context.get_input(2), element::i32));
+    auto src = context.get_input(3);
+    auto alpha = context.get_input(4);
+    auto converted_alpha = context.mark_node(std::make_shared<v1::ConvertLike>(alpha, src));
+    auto alpha_src = context.mark_node(std::make_shared<v1::Multiply>(converted_alpha, src));
+    auto input_shape_rank = get_shape_rank(context, input);
+    auto const_one = context.mark_node(v0::Constant::create(element::i32, Shape{1}, {1}));
+    auto const_one_0d = context.mark_node(v0::Constant::create(element::i32, Shape{}, {1}));
+    auto inp_rank = std::get<1>(input_shape_rank);
+    // ScatterElementsUpdate required that index, source and update have the same rank
+    // in aten::index_add index represents as 1d-array for specific dim and update may have different size
+    // from source in non-indexing axes
```

Comment:
> ...update may have different size
> // from source in non-indexing axes

@eaidova How can we get such a "update" input tensor? According to https://pytorch.org/docs/stable/generated/torch.Tensor.index_add_.html, there is such a description to "update" tensor.
> and all other dimensions must match self, or an error will be raised.

Reply:
description is not always correct and matching with expected behaviour (or can be outdated and not aligned with torch behaviour), on practice it can have larger rank

Reply:
> on practice it can have larger rank
a larger rank of aten "source"? We cannot broadcast from a larger rank tensor to a smaller one, can we?

Reply:
yes, larger then source rank and as the result, it is expected that rank stayed the same like for source